### PR TITLE
Race condition in ManualTimerExampleSpec fixed

### LINKED ISSUE #24957: Failed: ManualTimerExampleSpec
https://jenkins.akka.io:8498/job/akka-nightly-2.12/547/testReport/junit/akka.testkit.typed.scaladsl/ManualTimerExampleSpec/A_timer_must_replace_timer/

```
 java.lang.AssertionError: assertion failed: received unexpected message Tock(2)
	at scala.Predef$.assert(Predef.scala:219)
	at akka.testkit.typed.internal.TestProbeImpl.expectNoMessage_internal(TestProbeImpl.scala:172)
	at akka.testkit.typed.internal.TestProbeImpl.expectNoMessage(TestProbeImpl.scala:162)
	at akka.testkit.typed.scaladsl.ManualTime.$anonfun$expectNoMessageFor$1(ManualTime.scala:55)
	at akka.testkit.typed.scaladsl.ManualTime.$anonfun$expectNoMessageFor$1$adapted(ManualTime.scala:55)
	at akka.testkit.typed.scaladsl.ManualTime$$Lambda$530/2132707865.apply(Unknown Source)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:34)
	at akka.testkit.typed.scaladsl.ManualTime.expectNoMessageFor(ManualTime.scala:55)
	at akka.testkit.typed.scaladsl.ManualTimerExampleSpec.$anonfun$new$9(ManualTimerExampleSpec.scala:99)
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #24957

### CODE REVIEW

File: akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ManualTimerExampleSpec.scala [hunk 1]
```diff
@@ -83,21 +85,22 @@ class ManualTimerExampleSpec extends AbstractActorSpec {
...
       val ref = spawn(behavior)
-
       manualTime.timePasses(11.millis)
       probe.expectMessage(Tock(1))
 
-      // next Tock(1) enqueued in mailboxed, but should be discarded because of new timer
+      // next Tock(1) enqueued in mailbox, but should be discarded because of new timer
       ref ! SlowThenBump(2)
-      manualTime.expectNoMessageFor(interval, probe)
+      probe.expectMessage(SlowThenBumpAck)
```
Comment (johanandren) [inline akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ManualTimerExampleSpec.scala]: Race condition was between the next line (which advances the timer) and the timer advancement in the actor.

**Review by patriknw — APPROVED** [review]
LGTM

**Review by ktoso — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
PLS BUILD

**Pull request merged**